cumulusci/robotframework/salesforce_robot_library_base.py

```diff
@@ -0,0 +1,42 @@
+import logging
+
+from cumulusci.core.config import TaskConfig
+from robot.libraries.BuiltIn import BuiltIn
+from cumulusci.tasks.apex.anon import AnonymousApexTask
+
+from cumulusci.tasks.apex.batch import BatchApexWait
+
+
+class SalesforceRobotLibraryBase(object):
+    """Base class for any Robot Keyword Library which needs access to
+       Salesforce and CumulusCI features."""
+
+    logger = logging.getLogger("robot_debug")
+    logger.setLevel("DEBUG")
+
+    @property
+    def builtin(self):
+        return BuiltIn()
+
+    @property
+    def cumulusci(self):
+        """Get access to the CumulusCI Keyword Library"""
+        return self.builtin.get_library_instance("cumulusci.robotframework.CumulusCI")
+
+    @property
+    def salesforce(self):
+        """Get access to the Salesforce Keyword Library"""
+        return self.builtin.get_library_instance("cumulusci.robotframework.Salesforce")
+
+    def _run_subtask(self, taskclass, **options):
+        """Helper method for running CCI tasks"""
+        subtask_config = TaskConfig({"options": options})
+        return self.cumulusci._run_task(taskclass, subtask_config)
+
+    def _batch_apex_wait(self, class_name):
+        """Helper method for waiting for Salesforce jobs to finish"""
+        return self._run_subtask(BatchApexWait, class_name=class_name)
+
+    def _run_apex(self, code):
+        """Helper method for running Apex code easily"""
+        return self._run_subtask(AnonymousApexTask, apex=code)
```

Comment:
They are intended for use by Python programmers (e.g. me). 

run_task_class takes a string as an argument, which is appropriate for a Robot programmer. 

_run_subtask takes a class as an argument, which is appropriate for a Python programmer (and helps the programmer detect errors earlier/faster).

The other two could probably be keywords, if we aren't worried about keyword creep. Usually when you might be tempted to put Apex in a Robot script it might be better to make a new keyword in Python, so that's one argument for keeping that one as a helper function instead of a keyword. But on the other hand, maybe there are plentiful legitimate uses for Apex right in robot code.

Reply:
I agree passing a class is better so that typos can be caught at import time. But then my suggestion is to make run_task_class accept either a string or a class so that we don't need a new method that does subtly different things in a different location in the codebase. That's just going to confuse the next person who's trying to figure out the best way to run a task from Robot Framework.

Avoiding keyword creep is good. I think avoiding function creep is good too. These are one-liners so what is the benefit of factoring them into methods instead of inlining them? I think it just obscures the details of what's happening and makes it harder to read the code that calls these.

Reply:
I agree with David - if we're going to make a change like this, I think it makes more sense to define `run_task` to accept either a string or a dictionary.